neo/Consensus/ConsensusService.cs

```diff
@@ -299,20 +303,30 @@ private void OnRecoveryMessageReceived(ConsensusPayload payload, RecoveryMessage
                     {
                         ConsensusPayload prepareRequestPayload = message.GetPrepareRequestPayload(context, payload);
                         if (prepareRequestPayload != null)
-                            ReverifyAndProcessPayload(prepareRequestPayload);
+                        {
+                            totalPrepReq = 1;
+                            if (ReverifyAndProcessPayload(prepareRequestPayload)) validPrepReq++;
+                        }
                         else if (context.IsPrimary())
                             SendPrepareRequest();
                     }
                     ConsensusPayload[] prepareResponsePayloads = message.GetPrepareResponsePayloads(context, payload);
+                    totalPrepResponses = prepareResponsePayloads.Length;
                     foreach (ConsensusPayload prepareResponsePayload in prepareResponsePayloads)
-                        ReverifyAndProcessPayload(prepareResponsePayload);
+                        if (ReverifyAndProcessPayload(prepareResponsePayload)) validPrepResponses++;
                 }
                 ConsensusPayload[] commitPayloads = message.GetCommitPayloadsFromRecoveryMessage(context, payload);
+                totalCommits = commitPayloads.Length;
                 foreach (ConsensusPayload commitPayload in commitPayloads)
-                    ReverifyAndProcessPayload(commitPayload);
+                    if (ReverifyAndProcessPayload(commitPayload)) validCommits++;
             }
             finally
             {
+                Log($"{nameof(OnRecoveryMessageReceived)}: finished (valid/total) " +
+                    $"ChgView: {validChangeViews}/{totalChangeViews} " +
```

Comment:
`total` is not the count of CN?

Reply:
I think it is:
`ChangeViewMessages = LastChangeViewPayloads.Where(p => p != null).Select(p => RecoveryMessage.ChangeViewPayloadCompact.FromPayload(p)).Take(this.M()).ToDictionary(p => (int)p.ValidatorIndex),`

Reply:
* `total` is the total number of messages examined during this run of the recovery
* `valid` is the the number of those examined that were valid